spdmlib/src/crypto/x509v3.rs

```diff
@@ -257,6 +292,113 @@ fn check_public_key_info(data: &[u8]) -> SpdmResult<usize> {
     check_and_skip_common_sequence(data)
 }
 
+fn check_and_get_extensions(data: &[u8]) -> SpdmResult<(usize, &[u8])> {
+    let len = data.len();
+    if len < 1 || data[0] != ASN1_TAG_EXPLICIT_EXTENSION {
+        Ok((len, &data[0..]))
+    } else {
+        let (payload_length, bytes_consumed) = check_length(&data[1..])?;
+        if len < 1 + bytes_consumed + payload_length {
+            Err(SPDM_STATUS_VERIF_FAIL)
+        } else {
+            Ok((
+                1 + bytes_consumed + payload_length,
+                &data[1 + bytes_consumed..1 + bytes_consumed + payload_length],
+            ))
+        }
+    }
+}
+
+fn get_key_usage_value(data: &[u8]) -> SpdmResult<(bool, u8)> {
+    let len = data.len();
+    let mut key_usage_value_len = 0;
+    let mut key_usage_value_index: usize = 0;
+    if len < 5 {
+        Ok((false, 0x00))
+    } else {
+        let mut find_key_usage: bool = false;
+        for index in 0..len - 5 {
+            if object_identifiers_are_same(&data[index..index + 5], OID_KEY_USAGE) {
+                let (key_usage_bytes_consumed, key_usage) =
+                    check_and_get_common_tag(&data[index + 5..])?;
+                if key_usage[0] != ASN1_TAG_BIT_STRING {
+                    find_key_usage = false;
+                } else {
+                    find_key_usage = true;
+                    (key_usage_value_len, _) = check_length(&key_usage[1..])?;
+                    key_usage_value_index =
+                        index + 5 + key_usage_bytes_consumed - key_usage_value_len;
+                }
+                break;
+            } else {
+                continue;
+            }
+        }
+        if find_key_usage {
+            Ok((
+                true,
+                data[key_usage_value_index + key_usage_value_len - 1] as u8,
+            ))
+        } else {
+            Ok((false, 0x00))
+        }
+    }
+}
+
+fn check_extensions_spdm_oid(data: &[u8], is_leaf_cert: bool) -> bool {
+    let len = data.len();
+    let oid_len = 10;
+    let extensions_end_index_oid = len - oid_len;
+    let mut responder_auth_oid_find_success: bool = false;
+    let mut requester_auth_oid_find_success: bool = false;
+    let mut spdm_extension_oid_find_success: bool = false;
+    let mut hardware_identity_oid_find_success: bool = false;
+    for index in 0..extensions_end_index_oid {
```

Comment:
The check should be more detailed, because the scope of the extension is relatively large and there may be coincidences(the data is same as OID, but is not actual OID).

Reply:
Thank you Wenxing, I will add OID TAG check after detect the target OID.